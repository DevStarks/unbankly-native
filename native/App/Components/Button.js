import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Text, TouchableOpacity } from 'react-native'
import styles from './Styles/ButtonStyles'
import ExamplesRegistry from '../Services/ExamplesRegistry'

ExamplesRegistry.addComponentExample('Button', () =>
  <Button
    text='Example button'
    onPress={() => window.alert('Your button was pressed')}
  />
)

class Button extends Component {
  static propTypes = {
    text: PropTypes.string,
    textStyle: PropTypes.obj,
    onPress: PropTypes.func
  }

  render () {
    return (
      <TouchableOpacity
        style={[styles.button, this.props.style]}
        onPress={this.props.onPress}>

        <Text style={[styles.buttonText, this.props.textStyle]}>
          {this.props.text}
        </Text>
      </TouchableOpacity>
    )
  }
}

export default Button
