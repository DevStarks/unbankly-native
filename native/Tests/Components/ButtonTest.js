import 'react-native'
import React from 'react'
import Button from '../../App/Components/Button'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

test('Button component renders correctly', () => {
  const tree = renderer.create(<Button onPress={() => {}} text='hi' />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('onPress', () => {
  let i = 0 // i guess i could have used sinon here too... less is more i guess
  const onPress = () => i++
  const wrapperPress = shallow(<Button onPress={onPress} text='hi' />)

  expect(wrapperPress.prop('onPress')).toBe(onPress) // uses the right handler
  expect(i).toBe(0)
  wrapperPress.simulate('press')
  expect(i).toBe(1)
})
