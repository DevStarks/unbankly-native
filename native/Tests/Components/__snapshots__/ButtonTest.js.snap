// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#A9CEC9",
      "opacity": 1,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#637381",
          "fontFamily": "Oswald-Bold",
          "fontSize": 20,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    hi
  </Text>
</View>
`;
